yamlcrd/pkg/parser/testdata/simple1/input.yaml

```diff
@@ -0,0 +1,22 @@
+XNamespace:
+  spec:
+    # labels is some labels
+    labels: map(key=string,value=string)
+
+    # instanceType is an instanceType
+    instanceType: string
+
+    # minReplicas is the minimum number of replicas
+    minReplicas: int(min=0)
+
+    # maxReplicas is the maximum number of replicas
+    maxReplicas: int(min=0)
```

Comment:
I'm assuming the `int(min=0)` language is entirely made-up? What are some of the principles behind it?

Reply:
Inspired by your kabstract python syntax.  I think that named parameters are much better than positional ones here.  We don't have anything particularly relevant in yaml.  I guess that maybe we could do something like `int{ min: 0 }` but I think that might actually interact relatively poorly with yaml - I'm trying to do things that aren't too painful with yaml escaping rules etc.

I guess one big question is what about string values, how will quoting work there - I guess I'll try adding a string or two :-)